api/main_endpoints/routes/DoorCode.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const router = express.Router();
+const DoorCode = require('../models/DoorCode');
+const {
+  checkIfTokenSent,
+  checkIfTokenValid,
+} = require('../util/token-functions');
+const {
+  OK,
+  BAD_REQUEST,
+  UNAUTHORIZED,
+  FORBIDDEN,
+  NOT_FOUND,
+} = require('../../util/constants').STATUS_CODES;
+const addErrorLog = require('../util/logging-helpers');
+
+router.get('/getAvailableCodes', (req, res) => {
+  DoorCode.find({ usersAssigned: { $lt: 2 } }).then((doorcodes) =>
+    res.status(OK).send(doorcodes)
+  );
+});
+
+router.post('/addCode', (req, res) => {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(FORBIDDEN);
+  } else if (!checkIfTokenValid(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+
+  const newCode = new DoorCode({
+    doorCode: req.body.doorCode,
+    doorCodeValidUntil: req.body.doorCodeValidUntil,
+    usersAssigned: req.body.usersAssigned,
+  });
+
+  DoorCode.create(newCode, (error, post) => {
+    if (error) {
+      return res.sendStatus(BAD_REQUEST);
+    }
+    return res.status(OK).send(post);
+  });
+});
+
+router.post('/editCode', (req, res) => {
```

Comment:
would we call on this endpoint for when we are assigning a user to a doorcode?

Reply:
the editCode endpoint is so that we can edit the usersAssigned if needed and this routing was created in case the attributes of a door code needed to be changed. Do you think it should be removed?